src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py

```diff
@@ -42,21 +46,43 @@ def __init__(self, span: SpanEntity, collection_id: str, created_by: dict) -> No
         self.collection_id = collection_id
         self.id = span.span_id
         self.created_by = created_by
+        self.external_event_data_uris = []
 
-    def persist(self, client):
+    def persist(self, cosmos_client: ContainerProxy, blob_container_client: ContainerClient, blob_base_uri: str):
         if self.id is None or self.partition_key is None or self.resource is None:
             return
 
+        if self.events and blob_container_client is not None and blob_base_uri is not None:
+            self._persist_events(blob_container_client, blob_base_uri)
+
         resource_attributes = self.resource.get(SpanFieldName.ATTRIBUTES, None)
         if resource_attributes is None:
             return
 
         from azure.cosmos.exceptions import CosmosResourceExistsError
```

Comment:
Shall we move this to top?

Reply:
I think we should not? We don't need to upload the data to blob when we do not persist span at cosmos db.

Reply:
Sorry for confusion. I intended to say the import line. Shall we align the style of importing?